src/java.base/share/native/libjli/java.h

```diff
@@ -51,23 +51,19 @@
 #define CURRENT_DATA_MODEL (CHAR_BIT * sizeof(void*))
 
 /*
- * The following environment variable is used to influence the behavior
- * of the jre exec'd through the SelectVersion routine.  The command line
- * options which specify the version are not passed to the exec'd version,
- * because that jre may be an older version which wouldn't recognize them.
- * This environment variable is known to this (and later) version and serves
- * to suppress the version selection code.  This is not only for efficiency,
- * but also for correctness, since any command line options have been
- * removed which would cause any value found in the manifest to be used.
- * This would be incorrect because the command line options are defined
- * to take precedence.
- *
- * The value associated with this environment variable is the MainClass
- * name from within the executable jar file (if any). This is strictly a
- * performance enhancement to avoid re-reading the jar file manifest.
- *
+ * Older versions of java launcher used to support JRE version selection - specifically,
+ * the java launcher in JDK 1.8 can be used to launch a java application using a different
+ * java runtime (older, newer or same version JRE installed at a different location) than
+ * the one the launcher belongs to.
+ * That support was discontinued starting Java 9. However, java launcher in JDK 1.8 can still
```

Comment:
I assume this should say "Java 9" rather than "JDK 9".

I read this paragraph a few times and one thing that I don't think is made clear is that it's the JDK 8 launcher that is exec'ing the JDK N launcher with the env variable set. Instead it speaks of launching the "higher version java runtimes". At some point we are going to have to fix an exit route. In the mean-time, anyone touching this code will read this and it may not be clear how the JDK N launcher gets exec'ed with this env variable set.

Reply:
Hello Alan, I've updated the PR to clarify this part of the comment. Let me know if the latest version reads better.

> At some point we are going to have to fix an exit route.

I suspect that exit route would have to come as a CSR in Java 8 to discontinue support for launching Java runtimes  greater than Java 23 (for example), through the mJRE selection mechanism.